File: rust/src/tests/string_test.rs
use crate::random::string::random_string;

struct TestStringStruct {
    length: usize,
}

#[test]
fn check_string() {
    let tests: [TestStringStruct; 4] = [
        TestStringStruct{ length: 5 },
        TestStringStruct{ length: 9 },
        TestStringStruct{ length: 15 },
        TestStringStruct{ length: 30 },
    ];

    for test in tests {
        let result = random_string(test.length);
        assert!(result.len() == test.length)
    }
}

File: golang/_test/string_test.go
package random

import (
	"fmt"
	"github.com/stretchr/testify/require"
	"random/random"
	"testing"
)

func TestRandomString(t *testing.T) {
	tests := []struct {
		name  string
		input int
	}{
		{
			name:  "_test 1",
			input: 5,
		},
		{
			name:  "_test 2",
			input: 9,
		},
		{
			name:  "_test 3",
			input: 15,
		},
		{
			name:  "_test 4",
			input: 30,
		},
	}

	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			result := random.String(test.input)
			require.Equal(t, test.input, len(result), fmt.Sprintf("length result must be same as input value"))
		})
	}
}

func BenchmarkRandomString(b *testing.B) {
	tests := []struct {
		name  string
		input int
	}{
		{
			name:  "_test 1",
			input: 5,
		},
		{
			name:  "_test 2",
			input: 9,
		},
		{
			name:  "_test 3",
			input: 15,
		},
		{
			name:  "_test 4",
			input: 30,
		},
	}
	for _, test := range tests {
		b.Run(test.name, func(b *testing.B) {
			for i := 0; i < b.N; i++ {
				random.String(test.input)
			}
		})
	}

}
